lib/improver/cube_combiner.py

```diff
@@ -80,6 +80,51 @@ def __repr__(self):
                                                self.warnings_on))
         return desc
 
+    @staticmethod
+    def expand_bounds(result_cube, cubelist, coord, point):
+        """Alter a coord such that bounds are expanded to cover
+        the entire range of the input cubes.
+
+        For example, in the case of time cubes if the input cubes have
+        bounds of [0000Z, 0100Z] & [0100Z, 0200Z] then the output cube will
+        have bounds of [0000Z,0200Z]
+
+        Args
+            result_cube (iris.cube.Cube):
+                A cube with metadata for the results.
+            cubelist (iris.cube.CubeList):
+                The list of cubes with coordinates to be combined
+            coord (str):
+                The coordinate to be combined.
+            point (str):
+                The method of calculating the new point for the coordinate.
+                Currently accepts:
+                    'mid' - halfway between the bounds
+                    'upper' - equal to the upper bound
+        Returns
+            result (iris.cube.Cube):
+                Cube with coord expanded.
+                n.b. If argument point == 'mid' then python will convert
+                result.coord('coord').points[0] to a float.
+        """
+        bounds = ([cube.coord(coord).bounds for cube in cubelist])
+        if None in bounds:
+            points = ([cube.coord(coord).points for cube in cubelist])
+            new_low_bound = np.min(points)
+            new_top_bound = np.max(points)
+        else:
+            new_low_bound = np.min(bounds)
+            new_top_bound = np.max(bounds)
+        result_cube.coord(coord).bounds = [[new_low_bound, new_top_bound]]
```

Comment:
There is an assumption here that the coord being modified only has a single point and a single set of bounds. I think at the least there needs to be a check and an error raised if there is more than one point.